### PR TITLE
Removed closing PHP line endings

### PR DESCRIPTION
**Pull request by jdarwood007**
These are useless really to have in our files.  A simple return after ?> causes mods adding to the end of that file to fail.  Their is no difference in how the software operates after removing these.

So lets just remove these guys and make everyone happier.

I used a regex to do this ([\s+]\?>\Z), so its possible that a few may of been missed.  I corrected db_last_error.php and Settings.php when SMF is modifying them.

### CODE REVIEW

**Comment by Fustrate** [comment]
IIRC, the package manager needs the ?> to add things at the end of the file. If you remove it, you'll need to fix the package manager.

**Comment by mikemill** [comment]
With the ?> gone it would be even easier now

```
file_put_contents($file, $code, FILE_APPEND);
```

**Comment by jdarwood007** [comment]
Fustrate,
PacMan handles it either way.

<pre>                   // Position the replacement at the end of the file (or just before the closing PHP tags).
                    elseif ($search['position'] === 'end')
                    {
                        if ($undo)
                        {
                            $actual_operation['searches'][$i]['preg_replace'] = '';
                        }
                        else
                        {
                            $actual_operation['searches'][$i]['preg_search'] = '(\\n\\?\\>)?$';
                            $actual_operation['searches'][$i]['preg_replace'] .= '$1';
                        }
                    }
</pre>

**Comment by Fustrate** [comment]
Ah, the last person I talked to about removing `?>` was under the impression that it depended on the presence of `?>`. Carry on.

**Comment by jdarwood007** [comment]
No problem, questions help ensure we on't make mistakes.

**Comment by emanuele45** [comment]
Can you also have a look at https://github.com/SimpleMachines/SMF2.1/issues/19 there are a couple of concerns about this.

Also it seems you chmodded all (or at least many) files from 644 to 755, is there a reason?

**Comment by jdarwood007** [comment]
Not sure why the chmod changed on them.  Thats how I got them when I setup the repo.  But a quick command and commit fixes that: find ./ -type f -exec chmod 0644 {} \;

I haven't seen problems with css files doing this in package manager.  Do we have any test cases to prove as such?

**Comment by emanuele45** [comment]
> I haven't seen problems with css files doing this in package manager. Do we have any test cases to prove as such?

Sorry, I was more thinking at Spuds's last comment:

> There are also quite a few hidden gems in the code along the lines of '?' . '>' and combos there of ... all of those will need to be found and modified for this to work.

**Comment by jdarwood007** [comment]
http://pastebin.com/UBD41rfc
Not too many more, those can be fixed up.

**Comment by jdarwood007** [comment]
FYI,  If this needs to wait until after the first release it can.  Would give more time to ensure its all done right.

**Comment by Fustrate** [comment]
By "first release", you mean "first alpha"?

**Comment by emanuele45** [comment]
TBH I'm not particularly fond of this change in this release (/me hates jumps in the dark and feels this is a kind of jump in the dark). But since it seems important to many people (3?) I would like to have at least:
1) the PR merge automatically (at the moment there are 6 or 7 conflicts),
2) fix the missing files,
3) [this](https://github.com/SimpleMachines/SMF2.1/issues/19#issuecomment-5561292) working properly.
And of course if it should end up in the final release it must be in the alpha (even more because it needs testing).

**Comment by jdarwood007** [comment]
Fustrate,
I am talking about after the "alpha" I guess you call it.  Since they want to release already, adding this in now doesn't seem like a good idea.  Letting it go in and then having time to iron them out more before another release sounds good.

Emanuele45,
1. I can do a update and fix the conflicts.
2. Missing files?  Did I miss some?  Which files?
3. I don't think this is an issue unless you are talking about boardmod and its need to use ?> to find the end of the file.  Otherwise XML has its own attribute that handles this nicely.  Any XML looking for ?> shouldn't be really.  If its a issue, we could add a combat fix for it, but really boardmod needs to go and XML searches should be using the proper search method.

**Comment by emanuele45** [comment]
1. yep the one added with the last commits. :P
2. I mean the fact that when you click on the details of the changes (while un/installing you get nothing in the popup. I'd like to see the changes as it is now (i.e. search for [code here] replace with [nothing here])

**Comment by Oldiesmann** [comment]
@jdarwood007 Can you fix the merge conflicts so we can get this in?

**Comment by emanuele45** [comment]
This PR was for the master branch, so at the moment is not into the release-2.1 branch, and so not in the most up to date code.
Sorry if I'm pointing out only now, but I didn't follow the discussion much.

File: Sources/Subs-Admin.php [hunk 1]
```diff
@@ -305,18 +305,16 @@ function updateSettingsFile($config_vars)
...
-			$end = $i;
 	}
 
-	// This should never happen, but apparently it is happening.
-	if (empty($end) || $end < 10)
-		$end = count($settingsArray) - 1;
-
 	// Still more variables to go?  Then lets add them at the end.
 	if (!empty($config_vars))
+<<<<<<< HEAD
```
Comment (XinYenFon) [inline Sources/Subs-Admin.php:312]: This and line 317 and line 329 can output error afaik.

**Comment by jdarwood007** [comment]
I guess its a good thing its on the master branch and not the 2.1 branch that I did the pull request for...

**Comment by emanuele45** [comment]
Considering the release-2.1 branch was created 10 months ago and you sent this PR 11 months ago, it is still where it was originally generated. ;) (And don't worry, you are not alone there are other 2 "important" PRs waiting on a completely outdated branch (and please note that both are not properly signed-off and one of the two authors already stated that he is not interested in signing it off, but luckily a third person can do that if there is interest). I tried once to raise some awareness but without much feedback, so I gave up.)

**Pull request merged**